### PR TITLE
improve pass pipeline

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

**Comment by MaxGraey** [comment]
Recent commits just try to measure optimization speed so don't pay attention

**Comment by MaxGraey** [comment]
last commit slightly increase size for most tests but significantly improve:
- std/array.optimized.wat `-185` lines and directize most of sort calls
- std/arraybuffer.optimized.wat `-21`
- std/new.optimized.wat `-21`
- std/typedarray.optimized.wat `-14`

**Comment by jtenner** [comment]
Wow that vacuum pass got rid of a lot.

**Comment by MaxGraey** [comment]
Nope. vacuum pass just remove nops which could affect to next passes and improve optimization speed. Overall all changes reduce `519` lines

**Comment by jtenner** [comment]
Maybe I misread somehow, I was just pointing out there were a lot of removed lines. Looks great.

**Comment by MaxGraey** [comment]
@dcodeIO it's ready for merge

File: tests/compiler/std/symbol.optimized.wat [hunk 1]
```diff
@@ -638,29 +640,34 @@
...
   i32.store
-  local.get $0
+  local.get $2
   local.get $1
   i32.store offset=4
-  local.get $0
+  local.get $4
+  local.tee $1
+  local.get $2
+  local.tee $3
```
Comment (dcodeIO) [inline tests/compiler/std/symbol.optimized.wat:659]: Do you have a theory on what's happening here?

File: tests/compiler/std/symbol.optimized.wat [hunk 2]
```diff
@@ -638,29 +640,34 @@
...
   i32.load offset=8
+  i32.ne
```
Comment (dcodeIO) [inline tests/compiler/std/symbol.optimized.wat:661]: While not adding much, dropping an `i32.ne` seems trivial, and I'd expect this to be dropped in any case.

File: tests/compiler/std/symbol.optimized.wat (hunk 1 above)
Comment (MaxGraey) [reply tests/compiler/std/symbol.optimized.wat:659]: std/symbol has most significant regress but I can't figure outing how it repair. May be you could try later

**Comment by MaxGraey** [comment]
Alright now `std/symbol` test even reduced

**Pull request merged**